zerver/lib/validator.py

```diff
@@ -418,6 +418,17 @@ def to_non_negative_int(s: str, max_int_size: int=2**32-1) -> int:
         raise ValueError('%s is too large (max %s)' % (x, max_int_size))
     return x
 
+def to_positive_or_allowed_int(allowed_integer: int) -> Callable[[str], int]:
+    @set_type_structure('int')
+    def convertor(s: str) -> int:
+        x = int(s)
+        if x is allowed_integer:
```

Comment:
This should be `x == allowed_integer`.  `is` should not be used with integers; while it happens to return True for two equal commonly used integers that Python pre-constructs a unique copy of, it produces wrong results with others:

```
>>> 1000 is 1000
True
>>> 100000 is 100000
True
>>> -12100000 is -12100000
False
```

Reply:
Oh, didn't know this before, thank you :)